extensions/parquet/table/src/main/java/io/deephaven/parquet/table/PartitionFormatter.java

```diff
@@ -0,0 +1,132 @@
+        }
+    },
+    ForBoolean {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Boolean) obj).toString();
+        }
+    },
+    ForChar {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Character) obj).toString();
+        }
+    },
+    ForByte {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Byte) obj).toString();
+        }
+    },
+    ForShort {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Short) obj).toString();
+        }
+    },
+    ForInt {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Integer) obj).toString();
+        }
+    },
+    ForLong {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Long) obj).toString();
+        }
+    },
+    ForFloat {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Float) obj).toString();
+        }
+    },
+    ForDouble {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Double) obj).toString();
+        }
+    },
+    ForBigInteger {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((BigInteger) obj).toString();
+        }
+    },
+    ForBigDecimal {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((BigDecimal) obj).toString();
+        }
+    },
+    ForInstant {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((Instant) obj).toString();
+        }
+    },
+    ForLocalDate {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return DateTimeUtils.formatDate((LocalDate) obj);
+        }
+    },
+    ForLocalTime {
+        @Override
+        public String format(@NotNull final Object obj) {
+            return ((LocalTime) obj).toString();
+        }
+    };
+
+    private static final Map<Class<?>, PartitionFormatter> typeMap = new HashMap<>();
+    static {
+        typeMap.put(String.class, ForString);
+        typeMap.put(Boolean.class, ForBoolean);
+        typeMap.put(Character.class, ForChar);
+        typeMap.put(Byte.class, ForByte);
+        typeMap.put(Short.class, ForShort);
+        typeMap.put(Integer.class, ForInt);
+        typeMap.put(Long.class, ForLong);
+        typeMap.put(Float.class, ForFloat);
+        typeMap.put(Double.class, ForDouble);
+        typeMap.put(BigInteger.class, ForBigInteger);
+        typeMap.put(BigDecimal.class, ForBigDecimal);
+        typeMap.put(Instant.class, ForInstant);
+        typeMap.put(LocalDate.class, ForLocalDate);
+        typeMap.put(LocalTime.class, ForLocalTime);
+    }
+
+    abstract String format(@NotNull Object obj);
```

Comment:
I'm not sure we shouldn't have some option for `null` partitioning column values *if* we can make them work symmetrically. Empty? `"null"`?